cpp/src/arrow/compute/kernels/scalar_string_test.cc

```diff
@@ -1757,6 +1758,24 @@ TYPED_TEST(TestStringKernels, Strptime) {
   std::string output1 = R"(["2020-05-01", null, "1900-12-11"])";
   StrptimeOptions options("%m/%d/%Y", TimeUnit::MICRO);
   this->CheckUnary("strptime", input1, timestamp(TimeUnit::MICRO), output1, &options);
+
+  input1 = R"(["5/1/2020 %z", null, "12/11/1900 %z"])";
+  options.format = "%m/%d/%Y %%z";
+  this->CheckUnary("strptime", input1, timestamp(TimeUnit::MICRO), output1, &options);
+}
+
+TYPED_TEST(TestStringKernels, StrptimeZoneOffset) {
+  if (!arrow::internal::kStrptimeSupportsZone) {
+    GTEST_SKIP() << "strptime does not support %z on this platform";
+  }
+  // N.B. BSD strptime only supports (+/-)HHMM and not the wider range
+  // of values GNU strptime supports.
+  std::string input1 = R"(["5/1/2020 +0100", null, "12/11/1900 -0130"])";
```

Comment:
Ah, so this actually already worked before as well, but the resulting type is different (timezone naive vs aware). So might still be worth checking that change in type explicitly, unless that is already covered elsewhere in the tests.

Reply:
A couple things here:
- BSD strptime doesn't support "Z" as noted in the docstring. It only supports the syntax here.
- The result type is based on the presence of a "%z" so that's covered here already.